src/main/java/org/opensearch/geospatial/index/mapper/xyshape/XYShapeFieldTypeParser.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.mapper.xyshape;
+
+import java.util.Map;
+
+import org.opensearch.index.mapper.AbstractShapeGeometryFieldMapper;
+import org.opensearch.index.mapper.MapperParsingException;
+
+// XYShapeFieldTypeParser to parse and validate mapping parameters
+public final class XYShapeFieldTypeParser extends AbstractShapeGeometryFieldMapper.TypeParser {
+    @Override
+    protected AbstractShapeGeometryFieldMapper.Builder newBuilder(String name, Map<String, Object> params) {
+        return new XYShapeFieldMapper.XYShapeFieldMapperBuilder(name);
+    }
+
+    @Override
+    protected boolean parseXContentParameters(String name, Map.Entry<String, Object> entry, Map<String, Object> params)
+        throws MapperParsingException {
+        // we don't have to parse for deprecated parameters
+        return false;
```

Comment:
please add what are the deprecated parameters here.

Reply:
This method is used by super class for GeoShape to parse parameters and compare with deprecated param list to identify if user passed any of the deprecated param list or not. This is not applicable for XYShape since we don't have any deprecated parameters

Reply:
Added context. This will result in unnecessary parameter check for deprecated parameters which doesn't exists.

Reply:
I believe we should not override it. Given the case it was added to support the older versions of GeoShapes we should be good, and we will never have the field in our context.

Reply:
Cool. I removed it.